addons/hr/models/hr_employee.py

```diff
@@ -351,9 +351,9 @@ def get_formview_id(self, access_uid=None):
         if access_uid:
             self_sudo = self.with_user(access_uid)
         else:
-            self_sudo = self
+            self_sudo = self.env.user
```

Comment:
does not seem needed to change that line. 🙂 

By doing that, you completely change the recordset contained `self_sudo` to give the current user. 🤔